trace/trace.go

```diff
@@ -472,3 +535,42 @@ func (s *Span) createChildSpan(ctx context.Context, async bool) (context.Context
 	ctx = PutSpanInContext(ctx, newSpan)
 	return ctx, newSpan
 }
+
+type idGenerator struct {
+	randomBytes   func([]byte) (int, error)
+	incrementLock sync.Mutex
+	nonRandomID   uint64
+}
```

Comment:
I'd like to rephrase that comment - there's some kind of generalization showing up here and I'm not sure why it's here. I think the task being fulfilled is "Please give me an ID of X length". Can we make the API for ID generation closer to that? Do we need generators or can this be a function call such as `trace.GetNewID(8)` to get an 8 character ID? Is that oversimplifying it in a way that will make other things more difficult? (I wonder about whether it's particularly useful to be able to use different ID generation in tests, for example.)

Reply:
Ooh, and that question about testing highlights a different change that might happen here. If we do want to be able to replace ID generation for testing, we'll want to use an interface here instead of a struct. And if we don't, there's room to simplify the calling code.